libcontainer/container_linux.go

```diff
@@ -112,28 +112,33 @@ func (c *Container) OCIState() (*specs.State, error) {
 	return c.currentOCIState()
 }
 
+// ignoreCgroupError filters out cgroup-related errors that can be ignored,
+// because the container is stopped and its cgroup is gone.
+func (c *Container) ignoreCgroupError(err error) error {
+	if err == nil {
+		return nil
+	}
+	if !errors.Is(err, os.ErrNotExist) {
+		return err
+	}
+	if c.runType() == Stopped && !c.cgroupManager.Exists() {
+		return nil
+	}
+	return err
+}
```

Comment:
Jus double checking, as the logic is a bit complex.

If the error _is_ an `os.errNotExist`, then we need to check the second part? (`if c.runType() == Stopped && !c.cgroupManager.Exists() {`) is that correct? (making sure the intent here was not `ErrNotExist` errors can be ignored).

Could flip the logic here as well to allow early return, and skipping `c.cgroupManager.Exists()`;

```go
if c.runType() != Stopped || c.cgroupManager.Exists() {
	return err
}

return nil
```

If the intent was to ignore `os.ErrNotExist`, then the first part could be;


```go
func (c *Container) ignoreCgroupError(err error) error {
 	if err == nil || errors.Is(err, os.ErrNotExist) {
 		return nil
 	}
```

Reply:
I modified the code to be (hopefully) more readable; PTAL